src/pages/signIn/SignIn.tsx

```diff
@@ -0,0 +1,122 @@
+import React, { SyntheticEvent, useState, useLayoutEffect } from 'react';
+
+import './SignInStyles.scss';
+import Input from '../../components/Input/Input';
+import Button from '../../components/Button/Button';
+import { config } from './config';
+import { Fields, ValidationFields, ConfigItem } from './types';
+
+const SignIn: React.FC = (): JSX.Element => {
+  const [success, setSuccess] = useState<Boolean>(false)
+  const [formFields, setFormFields] = useState<Fields>({})
+  const [isFieldsValid, setIsFieldsValid] = useState<ValidationFields>({})
+  const [isShowPassword, setShowPassword] = useState<Boolean>(true)
+
+  useLayoutEffect(() => {
+    let fields: Fields = {}
+    let validationVars: ValidationFields = {}
+    config.forEach((field) => {
+      const {id} = field
+      if(field.isRequired) {
+        fields = {
+          ...fields,
+          [id]: '',
+        }
+        validationVars = {
+          ...validationVars,
+          [id]: false,
+        }
+      }
+      else {
+        fields = {
+          ...fields,
+          [id]: '',
+        }
+      }
+    })
+    setFormFields(fields)
+    setIsFieldsValid(validationVars)
+  }, [])
+
+  const onChangeFieldsValue = (e: any) => {
```

Comment:
name onChangeFieldValue you change value for one field when you run this function